hazelcast/src/test/java/com/hazelcast/test/starter/ConfigConstructor.java

```diff
@@ -184,4 +185,17 @@ public static Object getValue(Object obj, String getter)
         Method method = obj.getClass().getMethod(getter, null);
         return method.invoke(obj, null);
     }
+
+    private static Properties copy(Properties original) {
+        if (original == null) {
+            return null;
+        }
+
+        Properties copy = new Properties();
```

Comment:
Actually it's hard to make an exact copy of a `Properties` object because the class also includes a `Properties default` object used as a fallback from `getProperty`. `Hashtable.putAll` will ignore the defaults. This fails:
```
        Properties defaults = new Properties();
        defaults.setProperty("key1", "value1");
        Properties original = new Properties(defaults);
        original.setProperty("key2", "value2");
        assertEquals("value1", original.getProperty("key1"));
        assertEquals("value2", original.getProperty("key2"));

        Properties copy = new Properties();
        copy.putAll(original);
        assertEquals(copy.getProperty("key2"), original.getProperty("key2"));
        assertEquals(copy.getProperty("key1"), original.getProperty("key1")); // <-- was not copied
```

`Properties.stringPropertyNames` will iterate over all property keys, taking defaults into account, so this ensures the copy's `getProperty` will have the same apparent behavior as the original object's.